File: Sources/Indigox.UUM.Naming/Model/NameStrategyDescriptor.cs
using System;
using Indigox.Common.Utilities;

namespace Indigox.UUM.Naming.Model
{
    public class NameStrategyDescriptor
    {
        public int ID { get; set; }
        public float Priority { get; set; }
        public bool Enabled { get; set; }
        public string Assembly { get; set; }
        public string ClassName { get; set; }
        public string Description { get; set; }
        public DateTime LastModifyTime { get; set; }

        public INameStrategy ConvertToNamestrategy()
        {
            return (INameStrategy)ReflectUtil.CreateInstance( this.ClassName + "," + this.Assembly );
        }
    }
}

File: Sources/Indigox.UUM.Application/js/Indigox/UUM/Application/NameStrategyDescriptor.js
define("Indigox/UUM/Application/NameStrategyDescriptor",
    [
        "Indigox.Web.JsLib.Models.RecordManager"
    ],
function (
    RecordManager
) {
    RecordManager.getInstance().register('NameStrategyDescriptor', {
        columns: [
            { name: 'ID', text: '编号', type: String },
            { name: 'Priority', text: '优先级', type: String },
            { name: 'Enabled', text: '是否启用', type: String },
            { name: 'Assembly', text: '程序集', type: String },
            { name: 'ClassName', text: '类名', type: String },
            { name: 'Description', text: '描述', type: String },
            { name: 'LastModifyTime', text: '最后修改时间', type: String }
        ],
        primaryKey: ['ID']
    });

});